dusty/compiler/spec_assembler.py

```diff
@@ -149,3 +149,16 @@ def get_all_repos(active_only=False, include_specs_repo=True):
         if spec['repo']:
             repos.add(Repo(spec['repo']))
     return repos
+
+def _get_dependent_repos_for_app_or_library(app_or_library_name):
+    specs = get_expanded_libs_specs()
+    spec = specs.get_app_or_lib(app_or_library_name)
+    repos = set()
+    for dependent_name in spec['depends']['libs']:
+        repos.add(Repo(specs.get_app_or_lib(dependent_name)['repo']))
+    return repos
+
+def get_all_repos_for_app_or_library(app_or_library_name):
```

Comment:
I think this is poorly name but is not just for tests.  It is more of, get_all_repos_for_app_or_library_container. If app1 depends on app2, it does not really care what repo app2 depends on, it just needs to wait for app2 to spin up before it spins up.  That is why I put them here.   What if i add _container to the end of thes function names?